torch/export/_trace.py

```diff
@@ -988,16 +972,32 @@ def forward(self, *args, **kwargs):
         fake_params_buffers = make_fake_params_buffers(
             fake_mode, _get_params_buffers(mod)
         )
+
         with fake_mode:
-            ep_non_strict = _export_non_strict(
-                mod,
-                fake_args,
-                fake_kwargs,
-                fake_params_buffers,
-                constant_attrs,
-                pre_dispatch=pre_dispatch,
-                transform=_tuplify_outputs,
-            )
+            with _fakify_script_objects(mod, fake_args, fake_kwargs, fake_mode) as (
+                patched_mod,
+                new_fake_args,
+                new_fake_kwargs,
+                new_fake_constant_attrs,
+                map_fake_to_real,
+            ):
```

Comment:
I don't really get why `_fakify_script_objects` returns new_fake_args and new_fake_kwargs -- are we saying that ScriptObjects can show up in both args and as constants and this is handling both?

Reply:
Yes, it looks like this is the case